packages/studio/src/components/Highlighters.tsx

```diff
@@ -0,0 +1,104 @@
+import { CSSProperties, useMemo } from "react";
+import { useComponentNames } from "../hooks/useActiveComponentName";
+import useStudioStore from "../store/useStudioStore";
+import { ComponentStateKind } from "@yext/studio-plugin";
+import DOMRectProperties from "../store/models/DOMRectProperties";
+
+/**
+ * Generates Highlighters for all selected components.
+ */
+export default function Highlighters(props: {
+  iframeEl: HTMLIFrameElement | null;
+}): JSX.Element {
+  const { iframeEl } = props;
+  const [selectedComponentUUIDs, selectedComponentRectsMap] = useStudioStore(
+    (store) => {
+      return [
+        store.pages.selectedComponentUUIDs,
+        store.pages.selectedComponentRectsMap,
+      ];
+    }
+  );
+
+  return (
+    <div>
+      {Array.from(selectedComponentUUIDs).map((uuid) => {
+        return (
+          <Highlighter
+            key={`${uuid}-key`}
+            uuid={uuid}
+            rect={selectedComponentRectsMap[uuid]}
+            iframeEl={iframeEl}
+          />
+        );
+      })}
+    </div>
+  );
+}
+
+/**
+ * Highlights the component in the page preview.
+ */
+function Highlighter({
+  uuid,
+  rect,
+  iframeEl,
+}: {
+  uuid: string;
+  rect: DOMRectProperties | undefined;
+  iframeEl: HTMLIFrameElement | null;
+}) {
+  const componentTree = useStudioStore((store) =>
```

Comment:
it looks like `componentTree` is only used to calculate `isErrorState` below? if so, can we get the `componentTree` directly from the store in that selector callback instead of subscribing to updates to it here?

Reply:
Done